examples/bq-email-exports/main.py

```diff
@@ -0,0 +1,181 @@
+                                          table.table_id))
+
+    # Run query on that table
+    job_config = bigquery.QueryJobConfig(destination=table_id)
+
+    # Define the SQL query
+    sql = """
+        SELECT
+        CONCAT(
+            'https://stackoverflow.com/questions/',
+            CAST(id as STRING)) as url,
+        view_count
+        FROM `bigquery-public-data.stackoverflow.posts_questions`
+        WHERE tags like '%google-bigquery%'
+        ORDER BY view_count DESC
+        LIMIT 10
+    """
+
+    # Start the query, passing in the extra configuration
+    query_job = bq_client.query(sql, job_config=job_config)
+
+    # Wait for the job to complete
+    query_job.result()
+    print("Query results loaded to the table {}".format(table_id))
+
+    # Export table data as CSV to GCS
+    destination_uri = "gs://{}/{}".format(bucket_name, csv_name)
+    dataset_ref = bq_client.dataset(dataset_id, project=project)
+    table_ref = dataset_ref.table(file_name)
+
+    extract_job = bq_client.extract_table(
+        table_ref,
+        destination_uri,
+        # Location must match that of the source table
+        location="US",
+    )
+
+    # Waits for job to complete
+    extract_job.result()
+    print("Exported {}:{}.{} to {}".format(project, dataset_id, table_id,
+                                           destination_uri))
+
+    # Generate a v4 signed URL for downloading a blob
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(csv_name)
+
+    signing_credentials = None
+    # If running on GCF, generate signing credentials
+    # Service account running the GCF must have Service Account Token Creator role
+    if os.getenv("IS_LOCAL") is None:
+        signer = iam.Signer(
+            request=requests.Request(),
+            credentials=credentials(),
+            service_account_email=os.getenv("FUNCTION_IDENTITY"),
+        )
+        # Create Token-based service account credentials for signing
+        signing_credentials = service_account.IDTokenCredentials(
+            signer=signer,
+            token_uri="https://www.googleapis.com/oauth2/v4/token",
+            target_audience="",
+            service_account_email=os.getenv("FUNCTION_IDENTITY"),
+        )
+
+    url = blob.generate_signed_url(
+        version="v4",
+        # This URL is valid for 24 hours, until the next email
+        expiration=datetime.timedelta(hours=24),
+        # Allow GET requests using this URL
+        method="GET",
+        # Signing credentials; if None falls back to json credentials in local environment
+        credentials=signing_credentials,
+    )
+    print("Generated GET signed URL.")
+
+    # Create email message through SendGrid with link to signed URL
+    message = Mail(
+        from_email=from_email,
+        to_emails=to_emails,
+        subject="Daily BQ export",
+        html_content="<p> Your daily BigQuery export from Google Cloud Platform \
```

Comment:
Updated the email subject to be configurable. The message is a bit more complicated since the URL is embedded in the HTML, which doesn't support customizable messages. An option is using `plain_text_content= {customized message} + url` instead of the `html_content` but the long URL in the email looks really ugly...
What do you think?